code/controllers/subsystem/tts.dm

```diff
@@ -65,6 +67,23 @@ SUBSYSTEM_DEF(tts)
 		return FALSE
 	available_speakers = json_decode(response.body)
 	tts_enabled = TRUE
+	if(CONFIG_GET(str_list/tts_voice_blacklist))
+		var/list/blacklisted_voices = CONFIG_GET(str_list/tts_voice_blacklist)
+		log_config("Processing the TTS voice blacklist.")
+		for(var/voice in blacklisted_voices)
+			if(available_speakers.Find(voice))
+				log_config("Removed speaker [voice] from the TTS voice pool per config.")
+				available_speakers.Remove(voice)
+	var/datum/http_request/request_pitch = new()
+	var/list/headers_pitch = list()
+	headers_pitch["Authorization"] = CONFIG_GET(string/tts_http_token)
+	request_pitch.prepare(RUSTG_HTTP_METHOD_GET, "[CONFIG_GET(string/tts_http_url)]/pitch-available", "", headers_pitch)
+	request_pitch.begin_async()
+	UNTIL(request_pitch.is_complete())
+	pitch_enabled = TRUE
+	var/datum/http_response/response_pitch = request_pitch.into_response()
+	if(response_pitch.errored || response_pitch.status_code != 200)
+		pitch_enabled = FALSE
```

Comment:
throw a stacktrace on error here, if the http fails for any other reason than just straight up saying that tts unavailable.